src/Functions/UserDefined/UserDefinedSQLFunctionVisitor.cpp

```diff
@@ -86,22 +86,23 @@ void UserDefinedSQLFunctionVisitor::visit(ASTPtr & ast)
 
     if (auto * alter = ast->as<ASTAlterCommand>())
     {
-        visit_child_with_shared_ptr(alter->col_decl);
-        visit_child_with_shared_ptr(alter->column);
-        visit_child_with_shared_ptr(alter->partition);
-        visit_child_with_shared_ptr(alter->order_by);
-        visit_child_with_shared_ptr(alter->sample_by);
-        visit_child_with_shared_ptr(alter->index_decl);
-        visit_child_with_shared_ptr(alter->index);
-        visit_child_with_shared_ptr(alter->constraint_decl);
-        visit_child_with_shared_ptr(alter->constraint);
-        visit_child_with_shared_ptr(alter->projection_decl);
-        visit_child_with_shared_ptr(alter->projection);
-        visit_child_with_shared_ptr(alter->predicate);
-        visit_child_with_shared_ptr(alter->update_assignments);
-        visit_child_with_shared_ptr(alter->values);
-        visit_child_with_shared_ptr(alter->ttl);
-        visit_child_with_shared_ptr(alter->select);
+        const auto visit_child_update_parent = [&](ASTPtr & child)
```

Comment:
this lambda seems to be doing the same as `visit_child_with_shared_ptr`: `updatePointerToChild` as far as I see does the same as the for loop in `visit_child_with_shared_ptr`, so seem equal?

Reply:
It uses `updatePointerToChild` so it is different, but yes, it should be the same, this visitor should use `updatePointerToChild` always to update childrens instead of writing the same code again.

But it will require some other refactoring as well (some ASTs does not implements `forEachPointerToChild`), and I prefer not to mix here other things, since it has some refactoring already.

Reply:
Ok

Reply:
Let's write a comment about this near this lambda?

Reply:
OK, added.

Reply:
Actually `ASTColumnDeclaration` cannot use `updatePointerToChild` in this visitor, since it need to process only `ttl` and `default_expression` and not more (https://github.com/ClickHouse/ClickHouse/pull/43539), so I will leave it as-is